src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,113 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] array;
+    private int size;
+
+    public ArrayList() {
+        array = (T[]) new Object[DEFAULT_CAPACITY];
+        size = 0;
+    }
+
+    private void grow() {
+        Object[] newArray = new Object[array.length * 3 / 2];
+        System.arraycopy(array, 0, newArray, 0, array.length);
+        array = (T[]) newArray;
+
+    }
 
     @Override
     public void add(T value) {
-
+        add(value, size);
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index == size) {
+            if (size == array.length) {
+                grow();
+            }
+            array[size] = value;
+            size++;
+        } else if (index <= size - 1) {
+            size++;
+            if (size == array.length) {
+                grow();
+            }
+            System.arraycopy(array, index, array, index + 1,
+                    array.length - index - 1);
+            array[index] = value;
+        } else {
+            throw new ArrayIndexOutOfBoundsException("Array Index Out of bound " + index);
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            this.add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index >= 0 && index < size) {
+            return array[index];
+        } else {
+            throw new ArrayIndexOutOfBoundsException("Array Index Out of bound " + index);
+        }
     }
 
     @Override
     public void set(T value, int index) {
-
+        if (index >= 0 && index < size) {
```

Comment:
you are using this check in three places, let's make a separate method(boolean) out of this

Reply:
create method